.github/workflows/automated-release.yml

```diff
@@ -0,0 +1,116 @@
+name: Automated Release
+on:
+  release:
+    types: [published]
+jobs:
+  # release for linux
+  build-and-release-on-linux:
+    env:
+      TAG: ${{ github.ref_name }}
+      USER: merico-dev
+      REPO: stream
+      PLUGINDIR: .devstream
+      GOOS: linux
+      GOARCH: amd64
+    runs-on: ubuntu-latest
+    steps:
+      - run: echo "🐧 This job is now running on a ${{ runner.os }}-${{ runner.arch }} server hosted by GitHub!"
+      - name: Checkout
+        uses: actions/checkout@v2
+      - name: Setup Golang env
+        uses: actions/setup-go@v2
+        with:
+          go-version: 1.17
+      - name: Cache
+        uses: actions/cache@v2
+        with:
+          path: |
+            ~/.cache/go-build
+            ~/go/pkg/mod
+          key: ${{ runner.os }}-go-${{ hashFiles('**/go.sum') }}
+          restore-keys: |
+            ${{ runner.os }}-go-
+      - name: Build
+        run: make build
+      - name: Install Github-release
+        run: go install github.com/github-release/github-release@latest
+      - name: upload
+        run: |
+          #upload dtm
```

Comment:
I suggest we create a script for this part, so that it's easier to manage.

Reply:
ok, will refactor it